addons/point_of_sale/static/src/app/utils/input_popups/text_input_with_tag_popup.js

```diff
@@ -0,0 +1,102 @@
+import { Component, onMounted, useRef, useState } from "@odoo/owl";
+import { Dialog } from "@web/core/dialog/dialog";
+
+export class TextInputWithTagPopup extends Component {
```

Comment:
you can just add the new things to the existing text input component. No need to create a new one